modules/oas-page-builder/src/services/pageBuilder.ts

```diff
@@ -50,39 +59,80 @@ const getAtlasSpecUrl = async (apiKeyword: string) => {
   return oasFileURL;
 };
 
+interface GetOASpecParams {
+  sourceType: string;
+  source: string;
+  output: string;
+  pageSlug: string;
+  repoPath: string;
+  redocExecutor: RedocExecutor;
+  apiVersion?: string;
+  resourceVersion?: string;
+}
+
+async function getOASpec({
+  source,
+  sourceType,
+  repoPath,
+  pageSlug,
+  redocExecutor,
+  output,
+  apiVersion,
+  resourceVersion,
+}: GetOASpecParams) {
+  try {
+    let spec = '';
+    const buildOptions: RedocBuildOptions = {};
+    if (sourceType === 'url') {
+      spec = source;
+    } else if (sourceType === 'local') {
+      const localFilePath = normalizePath(`${repoPath}/source/${source}`);
+      spec = localFilePath;
+    } else if (sourceType === 'atlas') {
+      spec = await getAtlasSpecUrl({ apiKeyword: source, apiVersion, resourceVersion });
+      // Ignore "incompatible types" warnings for Atlas Admin API/cloud-docs
+
+      buildOptions['ignoreIncompatibleTypes'] = true;
+    } else {
+      throw new Error(`Unsupported source type "${sourceType}" for ${pageSlug}`);
+    }
+
+    const filePathExtension = `${apiVersion ? `/v${apiVersion.split('.')[0]}` : ''}${
```

Comment:
When attempting to build separate v1 and v2 pages locally, the resulting page includes the API version twice. Example:

```
/reference/api-resources-spec/v2/v2/index.html
```

I'm assuming it's because we're concatenating the API version to the file path here, but I'm wondering if anyone else is running into this or if this is just user error on my part 😅.

Reply:
I'm not having that problem when running solely the oas-page-builder module on cloud-docs. Let's sync about where you're seeing this.